python/ray/serve/tests/test_cancellation.py

```diff
@@ -365,5 +365,104 @@ async def get_out_of_band_response(self):
     assert h.get_out_of_band_response.remote().result() == "ok"
 
 
+def test_recursive_cancellation_during_execution(serve_instance):
```

Comment:
in order for these tests to be meaningful, don't we need to turn off the core cancellation support?

Reply:
So at first I took the approach of implementing full recursive cancellation at the Serve layer, but that leads to more of a performance drop. I think it's unnecessary to cover the part that core already covers (recursive task cancellation). We just need to take care of canceling the asyncio request assignment task before a request has been assigned to a replica.

So in this test file `test_recursive_cancellation_during_execution` already passes, and `test_recursive_cancellation_during_assignment` is the one that doesn't pass without this PR.